components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -2333,6 +2357,26 @@ private String doUserAuthorization(OAuthMessage oAuthMessage, String sessionData
 
         OAuth2Parameters oauth2Params = getOauth2Params(oAuthMessage);
         AuthenticatedUser authenticatedUser = getLoggedInUser(oAuthMessage);
+
+        // Here we validate all scopes, response type, client before consent page
+        HttpRequestHeaderHandler httpRequestHeaderHandler = new HttpRequestHeaderHandler(oAuthMessage.getRequest());
+        OAuth2AuthorizeReqDTO authzReqDTO =
+                buildAuthRequest(oauth2Params, oAuthMessage.getSessionDataCacheEntry(), httpRequestHeaderHandler);
+        try {
+            handleAuthorizationBeforeConsent(oAuthMessage, oauth2Params, authzReqDTO);
+        } catch (IdentityOAuth2ScopeValidationException e) {
```

Comment:
Does IdentityOAuth2ScopeValidationException give/cover all error scenarios that could be returned from handleAuthorizationBeforeConsent() method?

What i meant to ask is can only "scope validation" related exception be thrown from the handleAuthorizationBeforeConsent()  method?

Reply:
There are 3 errors return form handleAuthorizationBeforeConsent( ). `IdentityOAuth2ScopeValidationException `, `IdentityOAuth2Exception `, `InvalidOAuthClientException `.  here https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1950/files?diff=unified&w=0#diff-debebfb0b01b553afd77f73d24e1df466030d3d19b218e0d7b3005364a05939aR2434 we catch `IdentityOAuth2Exception `, `InvalidOAuthClientException ` and return as `OAuthSystemException `